kamon-jmx/src/main/scala/kamon/jmx/JMXReporter.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2015 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+*/
+
+package kamon.jmx
+
+import javax.management.{JMException, JMRuntimeException}
+
+import akka.actor._
+import akka.event.Logging
+import kamon.Kamon
+import kamon.metric.SubscriptionsDispatcher.TickMetricSnapshot
+
+object JMXReporter extends ExtensionId[JMXExtension] with ExtensionIdProvider {
+  override def lookup(): ExtensionId[_ <: Extension] = JMXReporter
+
+  override def createExtension(system: ExtendedActorSystem): JMXExtension = new JMXExtension(system)
+}
+
+class JMXExtension(system: ExtendedActorSystem) extends Kamon.Extension {
+  val log = Logging(system, getClass)
+  log.info("Starting the Kamon(JMX) extension")
+
+  val subscriber = system.actorOf(Props[JMXReporterSupervisor], "kamon-jmx-reporter")
+
+  Kamon.metrics.subscribe("trace", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("akka-actor", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("akka-dispatcher", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("counter", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("histogram", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("min-max-counter", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("gauge", "**", subscriber, permanently = true)
+  Kamon.metrics.subscribe("system-metric", "**", subscriber, permanently = true)
+}
+
+private trait ActorJMXSupervisor extends Actor with ActorLogging {
```

Comment:
@Tolsi instead of harcode the subscriptions we can to this:
```scala
 val subscriptions = jmxConfig.getConfig("subscriptions")

 subscriptions.firstLevelKeys foreach { subscriptionCategory ⇒
    subscriptions.getStringList(subscriptionCategory).asScala.foreach { pattern ⇒
      Kamon.metrics.subscribe(subscriptionCategory, pattern, subscriber, permanently = true)
    }
  }
```
where `subscriptions` is a key in the configuration file as above.